src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -57,13 +57,21 @@ public ActionResult execute() throws EntityDoesNotExistException {
             
             boolean isSectionChanged = student.isSectionChanged(originalStudentAttribute);
             boolean isTeamChanged = student.isTeamChanged(originalStudentAttribute);
+            boolean isEmailChanged = student.isEmailChanged(originalStudentAttribute);
+            
             if (isSectionChanged) {
                 logic.validateSectionsAndTeams(Arrays.asList(student), courseId);
             } else if (isTeamChanged) {
                 logic.validateTeams(Arrays.asList(student), courseId);
             }
             
             logic.updateStudent(studentEmail, student);
+            
+            if (isEmailChanged) {
+                logic.resetStudentGoogleId(student.email, courseId);
+                taskQueuer.scheduleCourseRegistrationInviteToStudent(courseId, student.email, true);
```

Comment:
I remember seeing an ongoing PR for sending emails when the email is changed. Currently, we don't send out any emails when the email is changed. Students rarely join courses but they need the updated link to submit responses or view results. Perhaps that can be done as a separate PR?